src/main/java/com/example/springjdbc/data/repository/BookRepository.java

```diff
@@ -0,0 +1,62 @@
+package com.example.springjdbc.data.repository;
+
+import com.example.springjdbc.dto.Book;
+import com.example.springjdbc.mapper.BookMapper;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@RequiredArgsConstructor
+public class BookRepository {
+
+    NamedParameterJdbcTemplate template;
+
+    //Old-fashioned way
+    JdbcTemplate jdbcTemplate;
+
+    public Long createBook(Book book) {
+        String sql = "INSERT INTO book (bookname, bookauthor) VALUES (:bookName, :bookAuthor) RETURNING ID";
+        SqlParameterSource sqlParameterSource = new MapSqlParameterSource()
+            .addValue("bookName", book.getBookName())
+            .addValue("bookAuthor", book.getBookAuthor());
+        return template.queryForObject(sql, sqlParameterSource, Long.class);
+    }
+
+    //PO JDBC
+    public Long createBookOldFashioned(Book book) {
```

Comment:
is not used

Reply:
yes, this method for early aproach via sql query with holders